lib/3scale/backend/stats/aggregator.rb

```diff
@@ -141,14 +141,23 @@ def update_alerts(transactions)
             transactions.group_by { |tx| tx.application_id }.each do |app_id, txs|
               service_id = txs.first.service_id # All the txs of an app belong to the same service
 
-              # The operations below are costly. They load all the usage limits
-              # and current usages to find the current utilization levels.
-              # That's why before that, we check if there are any alerts that
-              # can be raised.
+              # Finding the max utilization can be costly because it involves
+              # loading usage limits and current usages. That's why before that,
+              # we check if there are any alerts that can be raised.
               next unless Alerts.can_raise_more_alerts?(service_id, app_id)
 
               begin
-                max_utilization = Utilization.max_in_all_metrics(service_id, app_id)
+                max_utilization = if Alerts::UsagesChecked.need_to_check_all?(service_id, app_id)
+                                    max = Utilization.max_in_all_metrics(service_id, app_id)
+                                    Alerts::UsagesChecked.mark_all_checked(service_id, app_id)
+                                    max
```

Comment:
Great opportunity to `tap` a Ruby idiom:

```ruby
Utilization.max_in_all_metrics(service_id, app_id).tap do
  Alerts::UsagesChecked.mark_all_checked service_id, app_id
end
```

Reply:
Changed :+1: